src/hooks/useStatementOfCashFlow/useStatementOfCashFlow.tsx

```diff
@@ -1,72 +1,63 @@
-import { format, startOfDay } from 'date-fns'
 import useSWR from 'swr'
 import { useAuth } from '../useAuth'
 import { useEnvironment } from '../../providers/Environment/EnvironmentInputProvider'
-
-type UseStatementOfCashFlow = (
-  startDate?: Date,
-  endDate?: Date,
-) => {
-  data: StatementOfCashFlow | undefined
-  isLoading: boolean
-  error: unknown
-  refetch: () => void
+import { getStatementOfCashFlow } from '../../api/layer/statement-of-cash-flow'
+import { endOfMonth, startOfMonth } from 'date-fns'
+
+function buildKey({
+  access_token: accessToken,
+  apiUrl,
+  businessId,
+  startDate,
+  endDate,
+}: {
+  access_token?: string
+  apiUrl?: string
+  businessId: string
+  startDate: Date
+  endDate: Date
+}) {
+  if (accessToken && apiUrl) {
+    return {
+      accessToken,
+      apiUrl,
+      businessId,
+      startDate,
+      endDate,
+      tags: ['#statement-of-cash-flow'],
+    } as const
+  }
 }
 
-export const useStatementOfCashFlow: UseStatementOfCashFlow = (
-  startDate: Date = new Date(),
-  endDate: Date = new Date(),
-) => {
-  const { businessId, read, syncTimestamps, hasBeenTouched } =
-    useLayerContext()
-  const { apiUrl } = useEnvironment()
+export function useStatementOfCashFlow({
+  startDate = startOfMonth(new Date()),
+  endDate = endOfMonth(new Date()),
+}: {
+  startDate?: Date
+  endDate?: Date
+}) {
   const { data: auth } = useAuth()
-
-  const startDateString = format(
-    startOfDay(startDate),
-    'yyyy-MM-dd\'T\'HH:mm:ssXXX',
-  )
-  const endDateString = format(startOfDay(endDate), 'yyyy-MM-dd\'T\'HH:mm:ssXXX')
-
-  const queryKey =
-    businessId &&
-    startDateString &&
-    endDateString &&
-    auth?.access_token &&
-    `statement-of-cash-${businessId}-${startDateString}-${endDateString}`
-
-  const { data, isLoading, isValidating, error, mutate } = useSWR(
-    queryKey,
-    Layer.getStatementOfCashFlow(apiUrl, auth?.access_token, {
-      params: {
-        businessId,
-        startDate: startDateString,
-        endDate: endDateString,
-      },
+  const { apiUrl } = useEnvironment()
+  const { businessId } = useLayerContext()
+
+  return useSWR(
+    buildKey({
+      ...auth,
+      apiUrl,
+      businessId,
+      startDate,
+      endDate,
     }),
+    ({ apiUrl, accessToken, startDate, endDate }) => getStatementOfCashFlow(
+      apiUrl,
+      accessToken,
+      {
+        params: {
+          businessId,
+          startDate,
+          endDate,
+        },
+      })().then(({ data }) => data),
   )
-
-  const refetch = () => {
-    mutate()
-  }
-
-  // Refetch data if related models has been changed since last fetch
-  useEffect(() => {
-    if (queryKey && (isLoading || isValidating)) {
-      read(DataModel.STATEMENT_OF_CASH_FLOWS, queryKey)
-    }
-  }, [isLoading, isValidating])
-
-  useEffect(() => {
-    if (queryKey && hasBeenTouched(queryKey)) {
-      refetch()
-    }
-  }, [syncTimestamps, startDateString, endDateString])
```

Comment:
After removing this, the Statement of Cashflow won't be refetched automatically after ie. categorizing bank transaction. Here is the test case:

1. Visit Reports > Statement of cash flow and select ie. August 2024 - "remember" values from the Report
2. Go to Bank Transactions and categorize any transaction from August 2024
3. Go back to Reports > Statement of cash flow - without above code, SWR won't refetch data, so you will see same values as in step 1.
4. Refresh page - you should see new values